### PR TITLE
Fix deltaE cmc close colors bug

### LINKED ISSUE #1908: deltaE_cmc can return NaN for valid input values
Simple to reproduce:

```
import numpy as np
from skimage import color

c = np.array([0., 1.59607713, 0.87755709])
print(color.deltaE_cmc(c, c))
```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Fixes #1908 and closes #1909.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by rfezzani** [comment]
Thank you @jni

**Comment by hmaarrfk** [comment]
So i agree that this might address things in deltaE returning NaN, but it seems strange to pin this function to 0 in certain cases.


The challenge seems to be when computing `dE2` in, some pixels might actually return negative values.

I think you might as well use "clip" or an other formula that is guaranteed to be stable.

**Comment by hmaarrfk** [comment]
Yeah, I think you want to use `np.max(0, 2 * term)` to ensure that you always return a value above 0 for small roundoff errors.

**Comment by rfezzani** [comment]
You are right @hmaarrfk, I fixed it, but called `max` inside all the `sqrt` calls.

File: skimage/color/delta_e.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def deltaE_ciede94(lab1, lab2, kH=1, kC=1, kL=1, k1=0.045, k2=0.015):
     dE2 = (dL / (kL * SL)) ** 2
     dE2 += (dC / (kC * SC)) ** 2
     dE2 += dH2 / (kH * SH) ** 2
-    return np.sqrt(dE2)
+    return np.sqrt(np.maximum(dE2, 0))
```
Comment (hmaarrfk) [inline skimage/color/delta_e.py]:
```suggestion
    return np.sqrt(np.maximum(dE2, 0, maximum=dE2))
```

File: skimage/color/delta_e.py [hunk 2]
```diff
@@ -238,7 +238,7 @@ def deltaE_ciede2000(lab1, lab2, kL=1, kC=1, kH=1):
     dE2 += C_term ** 2
     dE2 += H_term ** 2
     dE2 += R_term
-    ans = np.sqrt(dE2)
+    ans = np.sqrt(np.maximum(0, dE2))
```
Comment (hmaarrfk) [inline skimage/color/delta_e.py]:
```suggestion
    return np.sqrt(np.maximum(dE2, 0, maximum=dE2))
```

File: skimage/color/delta_e.py [hunk 3]
```diff
@@ -304,7 +304,8 @@ def deltaE_cmc(lab1, lab2, kL=1, kC=1):
     dE2 = (dL / (kL * SL)) ** 2
     dE2 += (dC / (kC * SC)) ** 2
     dE2 += dH2 / (SH ** 2)
-    return np.sqrt(dE2)
+
+    return np.sqrt(np.maximum(0, dE2))
```
Comment (hmaarrfk) [inline skimage/color/delta_e.py]:
```suggestion
    return np.sqrt(np.maximum(dE2, 0, out=dE2))
```

**Comment by rfezzani** [comment]
I tried your suggestion @hmaarrfk, but it raises a `TypeError: return arrays must be of ArrayType` when a single color is provided :-(

**Comment by rfezzani** [comment]
Thank you @hmaarrfk, Are the added comments good enough?

**Comment by emmanuelle** [comment]
Merging, thanks @rfezzani !

**Comment by rfezzani** [comment]
Thank you @emmanuelle :tada:

**Pull request merged**